src/images/images.controller.ts

```diff
@@ -1,7 +1,81 @@
-import { Controller } from '@nestjs/common';
+import {
+  BadRequestException,
+  Controller,
+  Param,
+  ParseIntPipe,
+  Post,
+  UploadedFile,
+  UploadedFiles,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileFieldsInterceptor } from '@nestjs/platform-express/multer/interceptors/file-fields.interceptor';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { AuthGuard } from '@nestjs/passport';
+import {
+  PostFileUploadSwagger,
+  PostThumbnailUploadSwagger,
+  User,
+} from '../common/decorators';
 import { ImagesService } from './images.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
 
-@Controller('images')
+const imageType = [
+  'image/png',
+  'image/jpg',
+  'image/webp',
+  'image/jpeg',
+  'image/gif',
+];
+
+@ApiBearerAuth('accessToken')
+@ApiTags('image API')
+@Controller('uploads')
 export class ImagesController {
   constructor(private readonly imagesService: ImagesService) {}
+
+  @PostFileUploadSwagger()
+  @Post('/')
+  @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(
+    FileFieldsInterceptor([{ name: 'files', maxCount: 10 }], {
+      fileFilter: (req, file, callback) => {
+        const ext = file.mimetype;
+        if (!imageType.includes(ext)) {
+          callback(new BadRequestException('이미지 타입만 가능'), false);
+        }
+        callback(null, true);
+      },
+    }),
+  )
+  uploadFile(@UploadedFiles() files: { files: Express.MulterS3.File[] }) {
+    return this.imagesService.uploadFile(files);
+  }
```

Comment:
```suggestion
  uploadFile(@UploadedFiles() { files }: { files: Express.MulterS3.File[] }) {
    return this.imagesService.uploadFile(files);
  }
```

매개변수 files를 받는데 서비스 로직을 보니 `files.files` 형식으로 사용하더라고요. 여기서 바로 구조분해할당으로 받아주면 이후 로직에서 댑스가 줄어들어 보기 좋을 것 같습니다!

Reply:
아.. 이부분은 files라는 key로 받아야되는거라 files 그대로 가야될 것 같아요
 원래는 files가 안붙어야되는데 붙어서 오길래 어쩔 수 없이 해줬습니다ㅠ

Reply:
하긴... 코멘트 남기면서도 이유가 있겠거니 하긴 했는데 그런 이슈가 있었군요..😢